doc/smart-switch/high-availability/eni-based-forwarding.md

```diff
@@ -0,0 +1,288 @@
+
+### Phase 2 ###
+
+- BFD sessions are created to local DPU or the remote DPU for faster reactivity to card level failures
+- Orchagent will switch between primary and secondary endpoint based on BFD status
+
+## Architecture Design ##
+
+### ACL Table Configuration ### 
+```
+{
+    "ACL_TABLE_TYPE": {
+        "ENI_REDIRECT": {
+            "MATCHES": [
+                "TUNNEL_VNI",
+                "DST_IP",
+                "DST_IPV6",
+                "INNER_SRC_MAC",
+                "INNER_DST_MAC",
+                "TUNNEL_TERM"
+            ],
+            "ACTIONS": [
+                "REDIRECT_ACTION",
+            ],
+            "BIND_POINTS": [
+                "PORT"
+            ]
+        }
+    },
+    "ACL_TABLE": {
+        "ENI": {
+            "STAGE": "INGRESS",
+            "TYPE": "ENI_REDIRECT",
+            "PORTS": [
+                "<Ingress front panel ports>"
+            ]
+        }
+    }
+}
+```
+### ACL Rules ### 
+
+Assume the following ENI attributes
+```
+MAC: aa:bb:cc:dd:ee:ff
+TUNNEL_VNI: 4000
+VIP: 1.1.1.1/32
+```
+
+**ACL Rule for outbound traffic**
+
+```
+{  
+    "ACL_RULE": {
+        "ENI:aa:bb:cc:ff:fe:dd:ee:ff:OUT0": {
+            "PRIORITY": "999",
+            "TUNNEL_VNI": "4000",
+            "DST_IP": "1.1.1.1/32",
+            "INNER_SRC_MAC": "aa:bb:cc:dd:ee:ff"
+            "REDIRECT": "<local/remote nexthop oid>"
+        }
+    }
+}
+```
+
+**ACL Rule for inbound traffic**
+
+```
+{  
+    "ACL_RULE": {
+        "ENI:aa:bb:cc:ff:fe:dd:ee:ff:IN0": {
+            "PRIORITY": "999",
+            "TUNNEL_VNI": "4000",
+            "DST_IP": "1.1.1.1/32",
+            "INNER_DST_MAC": "aa:bb:cc:dd:ee:ff"
+            "REDIRECT": "<local/remote nexthop oid>"
+        }
+    }
+}
+```
+
+### Handling path loops after Tunnel decap ### 
+
+During HA failover, the HA pair will end up in a transitional state that makes it ambiguous to the switch if it is active or backup.
+
+When the HA failover happens, the used-to-be active becomes standby, but the used-to-be standby is still unchanged.
+
+This state, although brief in time, may lead to congestion, and packet drops on a switch.
+
+![Tunnel Termination Problem](./images/tunn_term_problem.png)
+
+To solve this, ACL rules with high priority are added and the redirect should always be to local nexthop
+
+![Tunnel Termination Solution](./images/tunn_term_solution.png)
+
+**ACL Rule for outbound traffic with Tunnel Termination**
+
+```
+{  
+    "ACL_RULE": {
+        "ENI:aa:bb:cc:ff:fe:dd:ee:ff:OUT1": {
+            "PRIORITY": "9999",
+            "TUNNEL_VNI": "4000",
+            "DST_IP": "1.1.1.1/32",
+            "INNER_SRC_MAC": "aa:bb:cc:dd:ee:ff",
+            "TUNN_TERM": "true",
+            "REDIRECT": "<local nexthop oid>"
+        }
+    }
+}
+```
+
+**ACL Rule for inbound traffic with Tunnel Termination**
+
+```
+{
+    "ACL_RULE": {
+        "ENI:aa:bb:cc:ff:fe:dd:ee:ff:IN1": {
+            "PRIORITY": "9999",
```

Comment:
we need to change this priority to 9998 or something lower as well.

Reply:
Updated